src/Service.Host/client/src/components/__tests__/impbooks/Impbook.test.js

```diff
@@ -0,0 +1,281 @@
+        expect(screen.getByText('Import Book')).toBeInTheDocument();
+        expect(screen.getByText('Order Details')).toBeInTheDocument();
+        expect(screen.getByText('Post Entries')).toBeInTheDocument();
+        expect(screen.getByText('Comments')).toBeInTheDocument();
+    });
+
+    test('can edit fields because have right permission', () => {
+        expect(screen.getByLabelText('Total Import Value')).not.toBeDisabled();
+        expect(screen.getByLabelText('Parcel Number')).not.toBeDisabled();
+        expect(screen.getByLabelText('Total Import Value')).not.toBeDisabled();
+        expect(screen.getByLabelText('Number of Cartons')).not.toBeDisabled();
+        expect(screen.getByLabelText('Number of Pallets')).not.toBeDisabled();
+        expect(screen.getByLabelText('Weight')).not.toBeDisabled();
+    });
+
+    test('but should never be able to edit certain fields', () => {
+        expect(screen.getByLabelText('Supplier Country')).toBeDisabled();
+        expect(screen.getByLabelText('EC (EU) Member')).toBeDisabled();
+        expect(screen.getByLabelText('Total Invoice Value')).toBeDisabled();
+        expect(screen.getByLabelText('Import Book Id')).toBeDisabled();
+    });
+});
+
+describe('When dont have right privilege', () => {
+    beforeEach(() => {
+        render(
+            <ImportBook
+                editStatus="create"
+                applicationState={applicationState}
+                history={history}
+                addItem={addItem}
+                updateItem={updateItem}
+                setEditStatus={setEditStatus}
+                setSnackbarVisible={setSnackbarVisible}
+                privileges={['not-right.priv']}
+                loading={false}
+            />
+        );
+    });
+
+    test('cannot edit fields', () => {
+        expect(screen.getByLabelText('Parcel Number')).toBeDisabled();
+        expect(screen.getByLabelText('Total Import Value')).toBeDisabled();
+        expect(screen.getByLabelText('Total Invoice Value')).toBeDisabled();
+        expect(screen.getByLabelText('Number of Cartons')).toBeDisabled();
+        expect(screen.getByLabelText('Number of Pallets')).toBeDisabled();
+        expect(screen.getByLabelText('Weight')).toBeDisabled();
+    });
+});
+
+describe('When editing', () => {
+    beforeEach(() => {
+        render(
+            <ImportBook
+                editStatus="edit"
+                applicationState={applicationState}
+                history={history}
+                addItem={addItem}
+                updateItem={updateItem}
+                setEditStatus={setEditStatus}
+                setSnackbarVisible={setSnackbarVisible}
+                privileges={privileges}
+                loading={false}
+                item={item}
+            />
+        );
+    });
+
+    test('invoice values are shown', () => {
+        expect(screen.getByText('99.01')).toBeInTheDocument();
+        expect(screen.getByText('1400')).toBeInTheDocument();
```

Comment:
you could do 
<code>
expect(screen.GetByLabelText('blah').value).toBe('99.01')
</code>

if that works. Just cos that tests the correct field has the correct value, as opposed to just checking for the presence of the correct value. I guess in theory your current test might not pick up cases where fields have their values mixed up, if that makes sense

Reply:
Unfortunately they're values in table cells, so these ones don't have their own labels. I did think about the wrong field having the right value, but reckoned the tests being there in this form were better than not being there and couldn't think of a nice way to get to exactly the right ones. Probably could through some magic selection pain if we really wanted